crates/teloxide/src/dispatching/update_listeners/webhooks.rs

```diff
@@ -1,13 +1,20 @@
 //!
-use std::net::SocketAddr;
+use std::{net::SocketAddr, path::PathBuf};
 
 use crate::{requests::Requester, types::InputFile};
 
 /// Options related to setting up webhooks.
 #[must_use]
+#[derive(Clone)]
+pub enum Location {
+    IP(SocketAddr),
+    // filesystem path to unix socket
+    Path(PathBuf),
+}
```

Comment:
I don't think changes like the following are a good idea:
```diff
-    let addr = ([0, 0, 0, 0], port).into();
+    let addr = webhooks::Location::IP(([0, 0, 0, 0], port).into());
```
`Location` should at least implement `From<([u8; 4], u8)>` like (`SocketAddr`) and also `From<SocketAddr>`, `From<PathBuf>` and `From<&Path>`. Some of these can be handled by `derive(derive_more::From)`.